interfaces/acados_template/acados_template/acados_casadi_ocp_solver.py

```diff
@@ -964,3 +965,191 @@ def cost_get(self, stage_: int, field_: str) -> np.ndarray:
+            lb = ca.vertcat(self.bounds['lbx'][self.index_map['lam_bx_in_lam_w'][stage]],
+                            self.bounds['lbx'][self.index_map['lam_bu_in_lam_w'][stage]],
+                            self.bounds['lbg'][self.index_map['pi_in_lam_g'][stage]],
+                            self.bounds['lbg'][self.index_map['lam_gnl_in_lam_g'][stage]]).full().flatten()
+            ub = ca.vertcat(self.bounds['ubx'][self.index_map['lam_bx_in_lam_w'][stage]],
+                            self.bounds['ubx'][self.index_map['lam_bu_in_lam_w'][stage]],
+                            self.bounds['ubg'][self.index_map['pi_in_lam_g'][stage]],
+                            self.bounds['ubg'][self.index_map['lam_gnl_in_lam_g'][stage]]).full().flatten()
+        elif stage == self.ocp.dims.N:
+            constraints_value = np.concatenate((self.nlp_sol_w[self.index_map['lam_bx_in_lam_w'][stage]],
+                                                self.nlp_sol_g[self.index_map['lam_gnl_in_lam_g'][stage]])).flatten()
+            lambda_values = np.concatenate((self.nlp_sol_lam_w[self.index_map['lam_bx_in_lam_w'][stage]],
+                                            self.nlp_sol_lam_g[self.index_map['lam_gnl_in_lam_g'][stage]])).flatten()
+            lb = ca.vertcat(self.bounds['lbx'][self.index_map['lam_bx_in_lam_w'][stage]],
+                            self.bounds['lbg'][self.index_map['lam_gnl_in_lam_g'][stage]]).full().flatten()
+            ub = ca.vertcat(self.bounds['ubx'][self.index_map['lam_bx_in_lam_w'][stage]],
+                            self.bounds['ubg'][self.index_map['lam_gnl_in_lam_g'][stage]]).full().flatten()
+        return  constraints_value, lambda_values, lb, ub
+    
+    def get_constraints_indices(self, stage: int):
+        """ 
+        Get the indices of the constraints for a given stage.
+        This function distinguishes between inequality and equality constraints and returns their indices.
+        """
+        constraints_value, _, lb, ub = self.get_constraints_value(stage)
+        # distinguish between equality and inequality constraints
+        if stage == 0:
+            nbx = self.ocp.dims.nbx_0
+            nbu = self.ocp.dims.nbu
+        elif stage < self.ocp.dims.N:
+            nbx = self.ocp.dims.nbx
+            nbu = self.ocp.dims.nbu
+        elif stage == self.ocp.dims.N:
+            nbx = self.ocp.dims.nbx_e
+            nbu = 0
+
+        ineq_indices = []
+        eq_indices_x = []
+        eq_indices_constraints = []
+        
+        for i in range(len(lb)):
+            if lb[i] != ub[i]:
+                ineq_indices.append(i)
+            else:
+                #distinguish between equality in decision variables and in constraints
+                if i in range(nbx + nbu):
+                    eq_indices_x.append(i)
+                else:
+                    eq_indices_constraints.append(i)
```

Comment:
rename to `eq_indices_ca_g`? they are the indices of equalities within g in the casadi formulation as far as I understand.